tests/Microsoft.Maui.IntegrationTests/Utilities/ToolRunner.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Diagnostics;
+using System.Text;
+
+namespace Microsoft.Maui.IntegrationTests
+{
+	public static class ToolRunner
+	{
+		public static string Run(string tool, string args, out int exitCode,
+			string workingDirectory = "",
+			int timeoutInSeconds = 600)
+		{
+			var info = new ProcessStartInfo(tool, args);
+
+			if (Directory.Exists(workingDirectory))
+				info.WorkingDirectory = workingDirectory;
+
+			return Run(info, out exitCode, timeoutInSeconds);
+		}
+
+		public static string Run(ProcessStartInfo info, out int exitCode,
+			int timeoutInSeconds = 600)
+		{
+			var procOutput = new StringBuilder();
+			using (var p = new Process())
+			{
+				p.StartInfo = info;
+				TestContext.WriteLine($"[ToolRunner] Running: {p.StartInfo.FileName} {p.StartInfo.Arguments}");
+				p.StartInfo.CreateNoWindow = true;
+				p.StartInfo.UseShellExecute = false;
+				p.StartInfo.RedirectStandardOutput = true;
+				p.StartInfo.RedirectStandardError = true;
+				p.OutputDataReceived += (sender, o) =>
+				{
+					if (!string.IsNullOrEmpty(o?.Data))
+					{
+						procOutput.AppendLine(o.Data);
```

Comment:
I think these data received events can occur on different threads, so you need a lock to synchronize access to `procOutput`.

Reply:
Thanks, I've added a lock around the AppendLine calls.